File: Coding Sample/Java/Fundamentals/Concurrency/ThreadLocalTest.java
class ThreadLocalTest{

	static class PrintItem{

		//private static String text;
		private static ThreadLocal<String> textSlot = new ThreadLocal<>();

		public static void write(String value){
			//text = value;
			textSlot.set(value);
		}

		public static String read(){
			//return text;
			return textSlot.get();
		}
	}

	static class Printer{

		public static void print(int copies){
			for(int i = 1; i <= copies; ++i){
				System.out.printf("%d:%s from thread<%x>%n", i, PrintItem.read(), Thread.currentThread().hashCode());
				Worker.doWork(i);
			}
		}
	}

	public static void main(String[] args) throws Exception{
		Thread child = new Thread(new Runnable(){
			public void run(){
				PrintItem.write("monday");
				Printer.print(5);
			}
		});
		child.start();
		PrintItem.write("sunday");
		Printer.print(5);
	}
}

File: Coding Sample/MS.NET/Fundamentals/Concurrency/threadlocaltest.cs
using System;
using System.Threading;

static class PrintItem
{
	[ThreadStatic]
	private static string text;

	public static void Write(string value)
	{
		text = value;
	}

	public static string Read()
	{
		return text;
	}
}

static class Printer
{
	public static void Print(int copies)
	{
		for(int i = 1; i <= copies; ++i)
		{
			Console.WriteLine("{0}:{1} from thread<{2}>", i, PrintItem.Read(), Thread.CurrentThread.ManagedThreadId);
			Worker.DoWork(i);
		}
	}
}

static class Program
{
	public static void Main()
	{
		Thread child = new Thread(() =>
		{
			PrintItem.Write("Monday");
			Printer.Print(5);
		});
		child.Start();

		PrintItem.Write("Tuesday");
		Printer.Print(5);
	}
}